src/test/engineeffectsdelay_test.cpp

```diff
@@ -105,9 +105,9 @@ TEST_F(EngineEffectsDelayTest, WholeBufferDelay) {
 
     const CSAMPLE inputBuffer[] = {-100.0, 100.0, -99.0, 99.0};
     const CSAMPLE zeroBuffer[] = {0.0, 0.0, 0.0, 0.0};
-    const CSAMPLE firstExpectedResult[] = {-100.0, 75.0, -49.5, 24.75};
+    const CSAMPLE firstExpectedResult[] = {-50.0, 50.0, -49.5, 49.5};
```

Comment:
Having to change the data in tests to fix them is usually a sign that some observable behavior has changed. Can you explain why the change of the values is still correct?

Reply:
Yeah of course. I will start with the change in the `EngineEffectsDelay`.

1. In the version before, the ramping was used for samples chunk independent of the channel count, so the previous version sequence was [-100.0, 75.0, -49.5, 24.75]. Based on the change of use `SampleUtil::linearCrossfadeBuffersOut` the crossfade (in this case more ramping due to crossfade with zero "silence" samples) works in the I would call "stereo mode". So the new expected result is [-100.0, 100.0, -49.5, 49.5].

2. Due to the feature in the `RingDelayBuffer`, that the first chunk of written samples is faded-in due to avoid a "crackling" sound when crossing the border from zero "silence" samples to real data, the expected result should change based on this feature. So, the final result is [-50.0, 50.0, -49.5, 49.5].

Reply:
Ah yes, that makes total sense. Thank you for the explanation.